src/renderer/widgets/IconButton.tsx

```diff
@@ -0,0 +1,60 @@
+import { KeyboardEvent, useMemo } from 'react';
+import DeleteIcon from 'renderer/icons/DeleteIcon';
+import EditIcon from 'renderer/icons/EditIcon';
+import TrashIcon from 'renderer/icons/TrashIcon';
+
+export enum IconName {
+  EDIT = 'Edit Icon',
+  DELETE = 'Delete Icon',
+  TRASH = 'Trash Icon',
+}
+
+interface IIconButtonProps {
+  icon: IconName;
+  isDisabled: boolean;
+  className?: string;
+  handleClick: () => void;
+}
+
+const IconButton = ({
+  icon,
+  isDisabled,
+  className,
+  handleClick,
+}: IIconButtonProps) => {
+  // Helper for detecting use of the ENTER key
+  const onKeyUp = (e: KeyboardEvent) => {
+    if (e.code === 'Enter') {
+      handleClick();
+    }
+  };
+
+  const getIcon = useMemo(() => {
```

Comment:
Can we instead use the children prop instead?

Reply:
That's true, but on the other hand, I appreciate that this structure keeps the actual icon components separate from components that use them. This reduces the amount of nested elements in other components. 

Plus, by adding new icons into this file, it ensures that we *have* to give it a name which would then be applied as the `aria-label` for the button. I don't think the overhead of having to update this file is necessarily a bad thing if it can also be used as a template for keeping good accessibility habits.